src/Tmds.Ssh/UserAuthentication.SshAgentAuth.cs

```diff
@@ -0,0 +1,138 @@
+// This file is part of Tmds.Ssh which is released under MIT.
+// See file LICENSE for full license details.
+
+using System.Buffers;
+using System.Net;
+using Microsoft.Extensions.Logging;
+
+namespace Tmds.Ssh;
+
+partial class UserAuthentication
+{
+    public sealed class SshAgentAuth
+    {
+        public static async Task<AuthResult> TryAuthenticate(SshAgentCredentials credential, UserAuthContext context, SshConnectionInfo connectionInfo, ILogger<SshClient> logger, CancellationToken ct)
+        {
+            EndPoint? endPoint = SshAgent.DefaultEndPoint;
+            if (endPoint is null)
+            {
+                return AuthResult.Skipped;
+            }
+
+            using var sshAgent = new SshAgent(endPoint, context.SequencePool);
+
+            if (!await sshAgent.TryConnect(ct))
+            {
+                return AuthResult.Skipped;
+            }
+
+            List<SshAgent.Identity> keys = await sshAgent.RequestIdentitiesAsync(ct);
+            if (keys.Count == 0)
+            {
+                return AuthResult.Skipped;
+            }
+
+            bool acceptedAlgorithm = false;
+            foreach (var key in keys)
+            {
+                if (context.IsSkipPublicAuthKey(key.PublicKey))
+                {
+                    continue;
+                }
+
+                SequenceReader reader = new SequenceReader(key.PublicKey);
+                Name keyType = reader.ReadName();
+
+                // Note: for unknown key type AlgorithmsForKeyType returns the key type as an algorithm. This is mostly true.
+                // This enables to use algorithms supported by the SSH Agent that this library doesn't directly support.
```

Comment:
@jborean93 :point_up: this is nice, SSH Agent enables support for key types that aren't directly supported by the library.